packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -1912,6 +1912,32 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
     // If text is obscured, the entire sentence should be treated as one word.
     if (obscureText) {
       return TextSelection(baseOffset: 0, extentOffset: _plainText.length);
+    // If the word is a space, on mobile try to select the previous word
+    // instead.
+    } else if (text?.text != null && word.textInside(text!.text!) == ' '
```

Comment:
This should also apply if there's more than one space I think? And there're other types of whitespace characters.

Reply:
Tried on iOS, if there's a newline character in between the new spaces it gets included in the selection. I think we insert LRM/RLM between spaces when the text contains both RTL and LTR directions, so that should probably count too.

Reply:
I updated this to use _isWhitespace instead of checking for a hardcoded space character. I also implemented a while loop to look backwards until a non-whitespace character is found.

Also, playing around on Android I realized that Android does not do this behavior, it just selects the space like desktop!  Only selectable non-editable text has this behavior on Android, which I will handle later in my SelectableText issue.  So I updated the code to only do the previous word selection on iOS.